### DISCUSSION TITLE
Fetch Failed Error (Ollama)

### DISCUSSION THREAD

**Discussion by MdNaeemurRahman** [FlowiseAI/Flowise#2675] [Q&A]
I'm running Flowise in Docker and also running Ollama (Not in Docker). For test, i created a simple Flow however it seems like it can't recognize Ollama and rasing Fetch Failed Error. How can i solve this issue?

![image](https://github.com/FlowiseAI/Flowise/assets/58079832/4869c5a6-e195-4571-a028-8ec14e768bd2)

Note: I'm in Linux and i didn't make any changes on the default docker-compose.yml, .env (except removing example) file

**Comment by BenjaminChan1105** [comment]
1 ) You may miss the "Cache".
2) I got the same problem: "TypeError: Fetch Failed". It was solved when I re-install the Ollama.

**Comment by DIYlover** [comment]
@MdNaeemurRahman To fix your issue, you can simple replace "localhost" with "host.docked.internal" in "Base URL" field in ChatOllama box.  "host.docker.internal" is a special hostname provided by the Docker daemon, which allows containers to access the host machine's IP address and network services.

Q: Why flowise can't reaches your ollama instance via localhost?
A: Due to your flowise webapp is running in docker container but your Ollama is actually running in the host (same level to your docker engine), they both are running in kind of isolated network actually.  

Hope it helps!!!

**Reply by farbensan** [reply]
Didn't work for me.

**Reply by ngalatis** [reply]
be careful, there is a small typo on the first "host.docker.internal" mention here. Notice its docker not docke**d**
hope it helps.

**Reply by farbensan** [reply]
Thanks. I tried it, but it's still not working. "http://host.docker.internal:11434/"

**Reply by ckleung99** [reply]
It does work. Thousands of thank you

**Reply by KallidValden** [reply]
This worked for me too!  "http://host.docker.internal:11434/"

**Reply by ezmonty** [reply]
worked!

**Reply by johngoad** [reply]
Thank you all so much

**Comment by daniel7634** [comment]
The issue may be caused by the Ollama service only listening on localhost (127.0.0.1) and not on the Docker network interface. To allow Docker containers to connect, you might need to configure Ollama to listen on all interfaces (0.0.0.0). Here’s how to fix it:
	
1.	Stop the current Ollama service:

`sudo systemctl stop ollama`

2.	Restart Ollama with the OLLAMA_HOST environment variable set to 0.0.0.0:

`sudo OLLAMA_HOST=0.0.0.0 ollama serve`

After this change, your Docker containers should be able to connect to the Ollama service.

**Reply by nishin** [reply]
changing to http://127.0.0.1:11434 for the Ollama base URL worked. thanks for the help

**Reply by pjdm** [reply]
That worked for me win11 where was getting FETCH 404 errors with localhost:11434. Thanks

**Reply by javierlongoriaroche** [reply]
I tried this and didn't work for me, but with this I saw this log from Ollama: **[GIN] 2025/01/16 - 11:41:09 | 404 |     103.615µs |       127.0.0.1 | POST     "/api/embeddings"** and I went to Ollama API Docs to searchfor that endpoint and I found that the correct endpoint is [/api/embed](https://github.com/ollama/ollama/blob/main/docs/api.md#generate-embeddings) , so it seems a problem with the Ollama Embeddings node in Flowise.

**Reply by surenderdev** [reply]
Thanks @nishin, your suggestion worked on my Win machine having ollama and flowise installed locally; 
Relacing localhost with 127.0.0.1 did it.
